### ISSUE TITLE
SABnzbd not updating / sherpa was the problem

### ISSUE THREAD

**Issue by an3k** [OneCDOnly/sherpa#199]
Hi,

TL;DR: Because sherpa was NOT downloading the package from the online repository but kept using the locally cached version it prevented a clean installation of SABnzbd.

I finally wanted to update my SABnzbd since it always showed a notification about a new version. I forgot that it auto-updates itself when you restart SABnzbd. One reason was because it could not update itself. And starting (or restarting) SAB also always took a long time.

I went and ran `sherpa upgrade SABnzbd` and that also took a long time for "downloading" SABnzbd. Directly afterwards I ran `sherpa upgrade SABnzbd debug`. Pastebin is here https://pastebin.com/A0nb8vSz

SABnzbd restarted but now the webinterface showed an error saying "wrong sabyenc version, expected 4.0.0, got 5.4.2". Also the settings revealed that I still ran SABnzbd 3.5.1

I ran `pip install sabyenc3==4.0.2` to get SABnzbd up and running again but now after I restarted SABnzbd it showed an error saying "no sabyenc version found".

Since it was finding sabyenc with version 5.4.2 installed I ran `pip install sabyenc3==5.4.2` but after a restart SABnzbd again said it cannot find sabyenc.

In both cases running `python3 -c "import sabyenc3; print("sabyenc3.__version__") "` returned an error "module not found" but `pip list` showed that sabyenc was installed.

I got angry because of all this "package manager above a package manager ******".

I manually backed up my config directory (mv) and ran the two commands:

`sherpa clean SABNzbd`
`sherpa reinstall SABnzbd debug`

I checked the debug and saw warnings about 

> Ignoring invalid distribution -abyenc3 (/share/CACHEDEV1_DATA/.qpkg/Entware/lib/python3.10/site-packages)

See https://pastebin.com/a5gmgyeu

At this time, SABnzbd 3.6.0 was running perfectly fine and starting or restarting was quick as usual. To see what the Warning was about I `ls -la /share/CACHEDEV1_DATA/.qpkg/Entware/lib/python3.10/site-packages` and saw a "temporary" directory called 

> ~abyenc3-5.4.2.dist-info/

I simply `rm -rf` it and re-run

`sherpa clean SABNzbd`
`sherpa reinstall SABnzbd debug`

The first command caused an error "couldn't clean local repository" and the second command ran through perfectly fine. See https://pastebin.com/JuT5yj7E

Now SABnzbd is running perfectly fine and no warnings or errors anywhere.

**Comment by OneCDOnly** [comment]
Hi Ben. Sorry for the trouble. 

You're running a few commands unnecessarily though. A quick reference is shown below:

```
sherpa upgrade sab
```
... will upgrade the SABnzbd QPKG only (not the application).

```
sherpa clean sab
```
... will remove all application files cloned from the SABnzbd GitHub repo, and download them again.

```
sherpa reinstall sab
```
... will reinstall the SABnzbd QPKG over the top of the existing one. This will overwrite application files, but additional files created by the application will remain.

If you find something isn't working right, please create an issue here, and allow me to assist. Posting your sherpa log online is a good idea.

**Comment by an3k** [comment]
Hi OneCD,

well, the issue that `sherpa reinstall SABnzbd` wasn't fixing the issue. Only after I ran `sherpa clean SABnzbd` and then `sherpa reinstall SABnzbd` again the problem was solved. It shouldn't be that way.

Also, is that automatic updating of SABnzbd (checking its github repo) after every start of it caused by SABnzbd or by the QPKG? I really wish that would be disabled permanently and by default.

**Comment by OneCDOnly** [comment]
> well, the issue that `sherpa reinstall SABnzbd` wasn't fixing the issue. Only after I ran `sherpa clean SABnzbd` and then `sherpa reinstall SABnzbd` again the problem was solved. It shouldn't be that way.

Okiedoke, if you're able to debug the scripts and find the issue, it would really help me out. I can't fix things I can't see. ;)

> Also, is that automatic updating of SABnzbd (checking its github repo) after every start of it caused by SABnzbd or by the QPKG?

Yes, the service script for the SABnzbd QPKG automatically performs a `git pull` (effectively) before every SABnzbd startup. This way, we can ensure the user's SABnzbd has the latest repo files, and makes it relatively easy for the user to ensure their application is up-to-date. They can `stop` then `start` the SABnzbd QPKG via their QTS App Center without needing to drop to the CLI. 

>  I really wish that would be disabled permanently and by default.

Why so?

**Comment by OneCDOnly** [comment]
@an3k any update here? Can we close this issue? 

Cheers.

**Comment by an3k** [comment]
How to debug the scripts? I only could provide the logs and I have no clue why having the package in local cache prevents the scripts from asking github about a new version.

I wish it wouldn't auto-update because it's the second time the update was in my case not necessary but also caused troubles after it updated. I just hate auto-updating stuff. Windows, TVs, Apps on my phone. They often introduce bugs or a new UI and I don't need either.

**Comment by OneCDOnly** [comment]
Ah, if you're unable to debug the scripts, then I understand. Thanks anyway. 

The locally cached package does not prevent git from updating from GitHub. The locally cached package can become corrupt over time. I'm yet to find a reason for this or a way to prevent it. For now, we have the `clean` operation which clears the locally cached files, which causes them to be re-downloaded when next starting the application via the service script.

If the auto-updating isn't for you, maybe try @QNAP_Stephane's QSabNZBd3 QPKG instead?

**Comment by an3k** [comment]
Well, how would I have to debug them? Any procedure you use?

All I can say is that nothing helped ... until I ran `clean` followed by `reinstall`. Maybe the locally cached files required for getting the new files from github were corrupted and `clean` was the only way to solve that ... but how did they become corrupted in the first place?

**Comment by OneCDOnly** [comment]
> Well, how would I have to debug them? Any procedure you use?

It's not something I can teach you in 5 minutes. You'd need to be proficient in BASH scripting before attempting to debug a large scripted environment like that established by `sherpa`. 

>  ... but how did they become corrupted in the first place?

As already said, I don't know why this happens.

It looks like we've gone as far as we can here, so I'm closing this issue. Thanks for reporting your problem and solution.
